modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/StorageHashIndexDescriptor.java

```diff
@@ -93,7 +94,7 @@ public String toString() {
      * @param index Catalog index descriptor.
      */
     public StorageHashIndexDescriptor(CatalogTableDescriptor table, CatalogHashIndexDescriptor index) {
-        this(index.id(), extractIndexColumnsConfiguration(table, index), !index.isCreatedWithTable());
+        this(index.id(), extractIndexColumnsConfiguration(table, index), index.status() == CatalogIndexStatus.REGISTERED);
```

Comment:
@ibessonov could you please remind us, why we used to deduce this value from a flag in the Catalog? I remember we had a discussion and you insisted on this approach based on the simplicity of the rebalance algorithm.

Reply:
Yes. The main idea here was a consistency of the way you create the index storage instances on newly joined nodes. `index.status()` is a mutable thing. Imagine the following:
- you have a raft group with 3 nodes, but 1 is offline
- new table is created
- upload some data
- new index is created
- eventually, new index is fully built and AVAILABLE
- now you start the 3rd node

There are 2 ways to create an index storage, as you see in the constructor - index is not built and index is fully built. In our scenario index must be built, but it's already available. Index creation procedure is thus inconsistent between the nodes in replication group. Even if it doesn't cause any visible bad behavior, it is still a design flaw and we should avoid it.

Reply:
@ibessonov thanks for the explanation! I'd just want to clarify a few things...

As far as I see, the only difference is whether we rebuild index or not. And even if the index was fully built, and complete index tree was rebalanced (<-- I may be wrong in this assumption) to a newly joined node, we still will rebuild index locally just because it was created this way on other nodes. But this won't make the index tree exactly the same anyway.

Is my understanding correct?

Reply:
> we still will rebuild index locally just because it was created this way on other nodes

No, the index will be built only during replication

Reply:
so, it will be built anyway disregarding this flag, right?

Reply:
yes

Reply:
@sashapolo I have reverted this change (keep flag `isCreatedWithTable` in catalog index descriptor).